From 4b5ce178a7f01bdce985a848285259f745c129fb Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:29 -0400
Subject: [usb] xhci: replace 'xhci->cmd_ring->dequeue' with 'trb' in stop_ep cmd handler

Message-id: <1410887463-20674-78-git-send-email-dzickus@redhat.com>
Patchwork-id: 95144
O-Subject: [RHEL7 PATCH 077/591] xhci: replace 'xhci->cmd_ring->dequeue' with 'trb' in stop_ep cmd handler
Bugzilla: 1032371
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1032371

commit bc752bde108400c80735ef72987acbca0eecefda
Author: Xenia Ragiadakou <burzalodowa@gmail.com>
Date:   Mon Sep 9 13:29:59 2013 +0300

    xhci: replace 'xhci->cmd_ring->dequeue' with 'trb' in stop_ep cmd handler

    This patch replaces 'xhci->cmd_ring->dequeue' with 'trb', the address of
    the command TRB, since it is available to reduce line length.

    Signed-off-by: Xenia Ragiadakou <burzalodowa@gmail.com>
    Acked-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-ring.c b/drivers/usb/host/xhci-ring.c
index eeca17e..d2f9b90 100644
--- a/drivers/usb/host/xhci-ring.c
+++ b/drivers/usb/host/xhci-ring.c
@@ -779,10 +779,8 @@ static void xhci_handle_cmd_stop_ep(struct xhci_hcd *xhci,
 
  struct xhci_dequeue_state deq_state;
 
- if (unlikely(TRB_TO_SUSPEND_PORT(
-        le32_to_cpu(xhci->cmd_ring->dequeue->generic.field[3])))) {
-  slot_id = TRB_TO_SLOT_ID(
-   le32_to_cpu(xhci->cmd_ring->dequeue->generic.field[3]));
+ if (unlikely(TRB_TO_SUSPEND_PORT(le32_to_cpu(trb->generic.field[3])))) {
+  slot_id = TRB_TO_SLOT_ID(le32_to_cpu(trb->generic.field[3]));
   virt_dev = xhci->devs[slot_id];
   if (virt_dev)
    handle_cmd_in_cmd_wait_list(xhci, virt_dev,
-- 
1.7.1